torch/_inductor/lowering.py

```diff
@@ -1543,6 +1543,12 @@ def apply_constraint(arg, fx_arg):
 # fails accuracy on test_torch.py, and explicit fallback required to avoid warn=True on implicit
 make_fallback(aten.exponential.default, warn=False)
 
+if torch.version.hip is not None:
+    # ROCm specific fallback, perf issues are observed when registered
+    make_fallback(aten.miopen_batch_norm, warn=False)
+    # tl.reduce not available yet in ROCm's version of triton
+    make_fallback(aten.prod, warn=False)
```

Comment:
This makes CPU fallback to ATen as well.

Reply:
Thanks @peterbell10, this should be okay just a temporary measure for ROCm until we can get triton reduce enabled.

cc: @dllehr-amd

Reply:
I think it's better to check "and torch.cuda.is_available to only fallback on GPU" to minimize potential unexpected consequences.

Reply:
Untested, but this code should make it fallback only on ROCm and not CPU:

```python
fallback_prod = fallback_handler(aten.prod)

@register_lowering(aten.prod)
def (x, axis=None, keepdims=False, *, dtype=None):
    if if torch.version.hip is not None and x.get_device() == "cuda":
        # Fallback on ROCm since it doesn't support tl.reduce yet
        if dim is not None:
            return fallback_prod(x, axis=axis, keepdims=keepdims, dtype=dtype)
        else:
            assert keepdims == False
            return fallback_prod(x, dtype=dtype)

    # ... 
```

Reply:
@kit1980 @peterbell10 

Proposing we go with this approach, aten.prod may not be the only fallback we will require exclusively for ROCm, so I would prefer to explicitly list with make_fallback to easily track our support as opposed to modifying each op to not register on ROCm. 

**Proposing**
```
# ROCm specific fallback, perf issues are observed when registered
make_fallback(aten.miopen_batch_norm, warn=False)

if torch.version.hip is not None and torch.cuda.is_available():
    # tl.reduce support not available yet
    make_fallback(aten.prod, warn=False)
```

If we anticipate some issues with this approach then I'm happy to go the other way and edit prod directly

Reply:
How about a `@fallback_on_rocm()` decorator?

Reply:
> How about a `@fallback_on_rocm()` decorator?

At the moment I think it makes more sense to be consistent and stick with the explicit `make_fallback` approach as this will allow us to be flexible in the way we create the specific fallbacks for ROCm, retaining functionality of the layout_constraint, warn and potentially future arguments that may be added to make_fallback. 

Otherwise we would potentially have to create multiple decorators for these cases e.g. in the way `require_dense` and `require_contiguous` are used for `aten.cudnn_rnn` and `aten.cudnn_rnn_backward`  https://github.com/pytorch/pytorch/blob/eee46ef5ba466a362ad79a5b6836c0afb32f9ca4/torch/_inductor/lowering.py#L1373

Reply:
If CPU inductor performing worse when an AMD GPU is connected is acceptable then go for it. So long as you're aware of the trade-off.